liblwgeom/gserialized2.c

```diff
@@ -519,6 +526,36 @@ gserialized2_peek_gbox_p(const GSERIALIZED *g, GBOX *gbox)
 	return LW_FAILURE;
 }
 
+int
+gserialized2_peek_first_point(const GSERIALIZED *g, POINT4D *out_point)
+{
+	uint8_t *data_ptr = gserialized2_get_geometry_p(g);
+
+	uint32_t type = (((uint32_t *)data_ptr)[0]);
+	int isEmpty = (((uint32_t *)data_ptr)[1]) == 0;
+	if (type != POINTTYPE || isEmpty)
```

Comment:
this does not match function name. what if input is something else?

Reply:
Do you think `gserialized$_point_peek_first` is better?

Reply:
"first" means it looks at first if there are many. This one just breaks if there's more than one.
For #436 may be cool to calculate hash on first point instead of going into loops of expanding it into a box and then taking its center, so a function that takes first point of whatever it gets will be useful. It can also speed up ST_StartPoint.

Reply:
Getting the point of a POINTTYPE is "trivial", but getting the first point of any geometry isn't (since it might involve recursion).  I think I might start throwing an error if something other than a POINT is passed and add other geometry types when needed, but I have to think this through.